### PR TITLE
Fix unused parameters in the `iocore` functionality (SSL and QUIC related code)

### PR DESCRIPTION
**Pull request by freak82**
Fix unused parameters using `/* name ATS_UNUSED */` for totally unused parameters.
Fix unused parameters using `[[maybe_unused]]` for conditionally unused parameters.

This pull request is part of the effort for [removing](https://github.com/apache/trafficserver/issues/11377) the `-Wno-unused-parameter` warning suppression.

### CODE REVIEW

File: src/iocore/net/QUICNetProcessor.cc [hunk 1]
```diff
@@ -50,7 +50,7 @@ DbgCtl dbg_ctl_iocore_net_processor{"iocore_net_processor"};
 } // end anonymous namespace
 
 static void
-debug_log(const char *line, void *argp)
+debug_log(const char *line, void * /* argp ATS_UNUSED */)
```
Comment (JosiahWI) [inline src/iocore/net/QUICNetProcessor.cc:53]: Please remove the `argp` parameter.

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Looks good.

**Comment by JosiahWI** [comment]
[approve ci rocky]

File: src/iocore/net/QUICNetProcessor.cc [hunk 2]
```diff
@@ -50,7 +50,7 @@ DbgCtl dbg_ctl_iocore_net_processor{"iocore_net_processor"};
 } // end anonymous namespace
 
 static void
-debug_log(const char *line, void * /* argp ATS_UNUSED */)
+debug_log(const char *line)
```
Comment (JosiahWI) [inline src/iocore/net/QUICNetProcessor.cc:53]: I didn't read this carefully - this is actually used as a callback so it needs the parameter.

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: src/iocore/net/QUICNetProcessor.cc (hunk 2 above)
Comment (JosiahWI) [reply src/iocore/net/QUICNetProcessor.cc:53]: I'm sorry.

Comment (freak82) [reply src/iocore/net/QUICNetProcessor.cc:53]: I just saw the same after I pushed the changes.
Shouldn't have hurried this.
I'm about to revert it back.

**Review by JosiahWI — APPROVED** [review]
Thanks for the cleanup!

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**